pallets/sto/src/lib.rs

```diff
@@ -0,0 +1,333 @@
+    }
+);
+
+decl_error! {
+    /// Errors for the Settlement module.
+    pub enum Error for Module<T: Trait> {
+        /// Sender does not have required permissions
+        Unauthorized,
+        /// An arithmetic operation overflowed
+        Overflow,
+        /// Not enough tokens left for sale
+        InsufficientTokensRemaining,
+        /// Fundraiser not found
+        FundraiserNotFound,
+        /// Fundraiser is frozen
+        FundraiserFrozen,
+        // Interacting with a fundraiser past the end `Moment`.
+        FundraiserExpired,
+        // Interacting with a fundraiser before the start `Moment`.
+        FundraiserNotStated,
+        // Using an invalid venue
+        InvalidVenue,
+        // Using an invalid portfolio
+        InvalidPortfolio,
+    }
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as StoCapped {
+        /// All fundraisers that are currently running. (ticker, fundraiser_id) -> Fundraiser
+        Fundraisers get(fn fundraisers): double_map hasher(blake2_128_concat) Ticker, hasher(twox_64_concat) u64 => Fundraiser<T::Balance, T::Moment>;
+        /// Total fundraisers created for a token
+        FundraiserCount get(fn fundraiser_count): map hasher(twox_64_concat) Ticker => u64;
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        type Error = Error<T>;
+
+        fn deposit_event() = default;
+
+        /// Create a new offering. A fixed amount of pre-minted tokens are put up for sale at the specified tiered rate.
+        #[weight = 800_000_000]
+        pub fn create_fundraiser(
+            origin,
+            offering_portfolio: PortfolioId,
+            offering_asset: Ticker,
+            raising_asset: Ticker,
+            tiers: Vec<PriceTier<T::Balance>>,
+            venue_id: u64,
+            start: Option<T::Moment>,
+            end: Option<T::Moment>,
+        ) -> DispatchResult {
+            let sender = ensure_signed(origin)?;
+            let did = Context::current_identity_or::<Identity<T>>(&sender)?;
+            ensure!(T::Asset::primary_issuance_agent(&offering_asset) == did, Error::<T>::Unauthorized);
+
+            ensure!(VenueInfo::contains_key(venue_id), Error::<T>::InvalidVenue);
+            let venue = VenueInfo::get(venue_id);
+
+            ensure!(venue.creator == did, Error::<T>::InvalidVenue);
+            ensure!(venue.venue_type == VenueType::Sto, Error::<T>::InvalidVenue);
+
+            ensure!(offering_portfolio.did == did, Error::<T>::InvalidPortfolio);
+            ensure!(<PortfolioAssetBalances<T>>::contains_key(offering_portfolio, offering_asset), Error::<T>::InvalidPortfolio);
+            let asset_balance = <PortfolioAssetBalances<T>>::get(offering_portfolio, offering_asset);
+
+            let offering_amount: T::Balance = tiers
+                .iter()
+                .map(|t| t.amount)
+                .fold(0.into(), |x, total| total + x);
```

Comment:
```suggestion
                .sum();
```

Reply:
the trait `std::iter::Sum` is not implemented for `<T as polymesh_common_utilities::CommonTrait>::Balance`

Reply:
Seems like something we can & should implement? Open-coding this fold everytime seems suboptimal.